featurebyte/query_graph/sql/feature_historical.py

```diff
@@ -390,46 +629,48 @@ async def get_historical_features(
     # use a unique request table name
     request_id = session.generate_session_unique_id()
     request_table_name = f"{REQUEST_TABLE_NAME}_{request_id}"
-
-    # Generate SQL code that computes the features
     request_table_columns = observation_set.columns
-    sql_expr = get_historical_features_expr(
-        graph=graph,
-        nodes=nodes,
-        request_table_columns=request_table_columns,
-        serving_names_mapping=serving_names_mapping,
-        source_type=source_type,
-        request_table_name=request_table_name,
-        parent_serving_preparation=parent_serving_preparation,
-    )
+
+    # Process nodes in batches
+    node_groups = split_nodes(nodes, NUM_FEATURES_PER_QUERY)
 
     # Execute feature SQL code
-    await observation_set.register_as_request_table(session, request_table_name)
+    await observation_set.register_as_request_table(
+        session, request_table_name, add_row_index=len(node_groups) > 1
+    )
 
     # Compute tiles on demand if required
     if not is_feature_list_deployed:
-        await compute_tiles_on_demand(
-            session=session,
-            graph=graph,
-            nodes=nodes,
-            request_id=request_id,
-            request_table_name=request_table_name,
-            request_table_columns=request_table_columns,
-            serving_names_mapping=serving_names_mapping,
-            parent_serving_preparation=parent_serving_preparation,
-            progress_callback=progress_callback,
-        )
+        tic = time.time()
+        for _nodes in node_groups:
+            logger.debug(f"Checking and computing tiles on demand for {len(_nodes)} nodes")
```

Comment:
Currently we skip the check in the linter, but would be good to add new logging statements with dynamic values in `extra`